src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -6837,17 +6839,19 @@ final void TryProcessFeatureExtAck(boolean featureExtAckReceived) throws SQLServ
     }
 
     final void start() {
-        timerThread = new Thread(this);
-        timerThread.setDaemon(true);
-        timerThread.start();
+        //if ExecutorService is shutdown already, reset it
+        if(executor.isShutdown()){
+            executor = Executors.newCachedThreadPool();
```

Comment:
The executor does not create daemon Threads by default, thus the result may be a little different.

I would probably use a thread factory as a parameter that use daemon threads.